pkl-core/src/main/java/org/pkl/core/stdlib/json/ParserNodes.java

```diff
@@ -172,6 +173,11 @@ public void endObject(@Nullable EconomicMap<Object, ObjectMember> members) {
 
     @Override
     public void startObjectValue(@Nullable EconomicMap<Object, ObjectMember> members, String name) {
+      if (!useMapping && "default".equals(name)) {
+        // https://github.com/apple/pkl/issues/561
+        throw new ParseException(
```

Comment:
Instead of failing, an entry with string key `"default"` could be added to the dynamic object. Not sure which behavior is more appropriate.

Reply:
I'm not sure that's any less surprising to users than throwing an exception here, but it at least broadens the set of valid inputs rather than shrinks it. Would love some input from the core team on this.

Reply:
We considered introducing this behavior in the past (before we added the `useMapping` option to the parser).

I don't feel strongly about this either way; curious what @holzensp or @stackoverflow think.

Reply:
Throwing an exception is the more principled approach.
Having `default` be an entry feels very hacky. It also has a much worse discoverability: people who don't know this behavior may try calling `obj.default` and get errors or subtle bugs that are hard to debug.
Also the exception even tells you how to fix the problem.